webharness/harness.js

```diff
@@ -184,9 +255,29 @@ addEventListener("error", harness_error);
 
 // Called by tests via test.js.
 function test_finished() {
+  socket.emit('unsubscribe', testroom);
+  var clients_exited = new Promise((resolve, reject) => {
```

Comment:
I'm not entirely convinced that waiting for both clients to exit is a good idea. We have already a synchronization point when both clients join the room for the next test and publish their test_loaded message into the admin room. So I don't see much additional value, but extra complexity. (Note: leave the socket.emit() above in place - simply do not worry about if the 'unsubscribe' was successful or not.)